File: Python/Quick_Sort.py
import random

def quick_sort(arr):
    if len(arr) <= 1:
        return arr

    pivot = arr[len(arr) // 2]
    left = []
    mid = []
    right = []

    for num in arr:
        if num < pivot:
            left.append(num)
        elif num > pivot:
            right.append(num)
        else:
            mid.append(num)

    return quick_sort(left) + mid + quick_sort(right)

def main():
    arr = [random.randint(0, 100) for _ in range(10)]
    print(arr)
    arr = quick_sort(arr)
    print(arr)

if __name__ == "__main__":
    main()

File: Java/Quick_Sort.java

List<Integer> quickSort(List<Integer> arr) {
    if (arr.size() <= 1) return arr;

    int pivot = arr.get(arr.size() / 2);

    List<Integer> left = new ArrayList<>();
    List<Integer> mid = new ArrayList<>();
    List<Integer> right = new ArrayList<>();

    for (int num : arr) {
        if (num < pivot) left.add(num);
        else if (num > pivot) right.add(num);
        else mid.add(num);
    }

    List<Integer> result = new ArrayList<>();
    result.addAll(quickSort(left));
    result.addAll(mid);
    result.addAll(quickSort(right));

    return result;
}

void main() {
    
        List<Integer> arr = new Random()
            .ints(10, 0, 100)
            .boxed()
            .collect(Collectors.toCollection(ArrayList::new));
        
    IO.println(arr);

    arr = quickSort(arr);

    IO.println(arr);
}